### DISCUSSION TITLE
Access data in other CRSes than EPSG:4326 via OGC API

### DISCUSSION THREAD

**Discussion by MayteTDGeograma** [gopeg/issues-barriers#34] [General]
The only EPSG available for the publication of data through OGC API is 4326, while the INSPIRE compliance reference system is EPSG 4258.

It would be desirable, to improve the usability of the OGC API, to speed up the possibility of using EPSG 4258 in this application

**Comment by sMorrone** [comment]
INSPIRE is "evolving", meaning that (once the new INSPIRE IR have been officially published, which should be shortly) the CRS list will be updated allowing for more EPSG codes e.g., the Member States will be allowed to deliver also in their own National CRSs, if these have been previously approved by the MIG and published in the (upcoming) INSPIRE CRS registry - find more details at https://github.com/INSPIRE-MIF/helpdesk/blob/main/crs.md

**Comment by rociorj** [comment]
We found the same issue for the Pop Wildfire use case. Thank you for the information on how the issue of reference systems will evolve within ISNPIRE. In the meantime, in our use case we have solved this by publishing the datasets in the system that supports OGC API Features, EPSG: 4326

**Comment by sMorrone** [comment]
An additional note: the Good Practice "OGC API – Features as an INSPIRE download service" explicitly mentions that 
**"NOTE 1 The Web API shall return collections and features in the default OAPIF coordinate reference system (WGS 84 longitude and latitude). However, the enclosure link for bulk download could still provide access to a data set in a different CRS.** "

**Comment by AndersOstmanNovogit** [comment]
This issue is reported in section 4.2.8 in D4.3. The solution described by Rocio is also described.

**Reply by AndersOstmanNovogit** [reply]
Now in D4.3 section 4.4.1

**Comment by sMorrone** [comment]
We should consider however that simply adding the CRS parameter, requests to the API service can return data in whatever CRS.
As an example consider the below query to GO-DEPTH APIs:
https://geoserver.epsilon-italia.it/geoserver/ogc/features/collections/go-peg:GeologicUnits_subsurface_Po_Plain/items?f=application%2Fgeo%2Bjson&filter=gu_Id=1&limit=20&crs=EPSG:3034
and compare to same query with no CRS parameter (returning therefore geometries in WGS84) 
https://geoserver.epsilon-italia.it/geoserver/ogc/features/collections/go-peg:GeologicUnits_subsurface_Po_Plain/items?f=application%2Fgeo%2Bjson&filter=gu_Id=1&limit=20

**Comment by sMorrone** [comment]
Summing up GO-DEPTH experience: 
the harmonised geopackage file created with hale studio is in EPSG:3034 (as per user requirements).
When GeoServer publishes through API from the GeoPackage feature store, it automatically transforms in geographic coordinate system WGS84. However, users can ask for data in EPSG:3034 or whatever CRS ( among those allowed by our service) simply adding the CRS parameter to their request

**Reply by AndersOstmanNovogit** [reply]
Now added to the selected solution.
